src/RestApi/Controllers/ProductCategories.php

```diff
@@ -106,6 +106,24 @@ protected function check_permissions( $request, $context = 'read' ) {
 	 * @return \WP_REST_Response
 	 */
 	public function prepare_item_for_response( $item, $request ) {
+		$products_of_category = get_posts(
```

Comment:
This is certainly a bottleneck - gathering all data about all posts in the store.

I think a direct SQL query would be a better way to go. The posts table includes a `comment_count` column, so join posts to the terms able and do some sums?

Joining on taxonomies can be difficult to understand so as a start, this would be how to join posts to terms:

```
FROM {$wpdb->posts} AS posts
INNER JOIN {$wpdb->term_relationships} AS term_relationships ON posts.ID = term_relationships.object_id
INNER JOIN {$wpdb->term_taxonomy} AS term_taxonomy USING( term_taxonomy_id )
INNER JOIN {$wpdb->terms} AS terms USING( term_id )
```

Reply:
I battled with this a little bit, tell me what do you think about it know?